### PR TITLE
Added RMS threading https://www.thorlabs.com/newgrouppage9.cfm?object…

### PR DESCRIPTION
**Pull request by ccrome**
…group_id=4960

Hi there, I added a microscope objective lens thread called RMS.  I just tested it on a microscope objective and it works swimmingly.

### CODE REVIEW

**Comment by adrianschlatter** [comment]
Hey @ccrome 

Thanks for supporting threadlib! And I'm glad you got your RMS thread problem solved. Regarding addition to threadlib, we still have to tackle a hurdle, though: Any direct modifications to THREAD_TABLE.scad will - unfortunately - get lost upon build because that file is autogenerated.

To make the change permanent, we need to add the rules to generate the thread specs to the build system. This involves creating of design/RMS_thread.csv and design/RMS_thread.awk (see [here](docs/CreatingThreadSpecs.md)). Usually, such an .awk file will generate an entire family of threads. For rms, it will only add two threads (ext and int). This is very similar to the PCO-threads we already have: Would you have a look at design/PCO_thread.csv and design/PCO_thread.awk and try to use the same approach we have there?

In plain English what the PCO approach does is:

* put the thread specs into the .csv file (instead of into THREAD_TABLE.scad)
* write an .awk script that does nothing except print every line of the .csv file (except for lines starting with '#' which are considered comments)
* add a line to Makefile so that the new .awk file actually gets run

**Comment by ccrome** [comment]
perhaps I should have read the CONTRIBUTING.md file :)

I just pushed a new version.

-Caleb
